Tuition rules

CalvinFronda opened the pull request.

pswk1 committed:
Apply formatting changes

CalvinFronda committed:
feat:(adminPortalTabs): semi working

Not really working ATM, If we refresh on any of the tabs, you will get a blank screen. We need to handle a nested route within a material-ui tab.

CalvinFronda committed:
feat(adminTabs): admin portal tabs have direct routes

I've solve an issue where hard refreshing on an admin portal tab will redirect to a 404 page.

CalvinFronda committed:
fix(): removed object map replaced with array of routes

CalvinFronda committed:
Merge branch 'bulk-upload-modals' into tuition-rules

CalvinFronda committed:
Apply formatting changes

dhuangg commented on src/components/FeatureViews/AdminPortal/AdminPortal.js:
```diff
@@ -10,12 +10,35 @@ import './AdminPortal.scss';
 
+    const { match, history } = props;
+    const { params } = match;
+    const { page } = params;
```
```
const {
    match: { 
       params: { page },
    },
    history,
} = props;
```

dhuangg commented on src/components/FeatureViews/AdminPortal/AdminPortalTabs.js:
```diff
@@ -1,58 +1,93 @@
+                </div>
+            )}
+            {selectedTabIndex === 2 && (
+                <Router>
```
let's put each tab into its own component

dhuangg reviewed (changes requested):
yo dude. this is lit. <3

CalvinFronda committed:
chore(adminTabs): Added TabPanel to handle tab content

pswk1 committed:
feat(tuition rules): use topic query w/ active rule field

- EditTuitionRules renamed to SetTuitionRules
- TuitionRules queries course topics with new field of active rules. Each topic is rendered in tuition rules tab with its respective active rule count

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rules): topics and their rules

- Clicking on a topic in the tuition rules tab routes to SetTuitionRules component that lists rules based on course type.
- if there is no catch-all rule present user cannot add instructor-specific rules

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rules): main tuition rule page component

- when admin views topic's tuition rules for the first time, they will be prompted to create a catch-all tuition rule if there isn't one already.
- admin can do this in main tuition rule page (component only for now)

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rules): set default rule functionality

- admin can set a default rule in MainTuitionRulePage. the mutation successfully posts to the backend

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
chore(tuition rules): move files to dedicated directory

- all tuition rules files placed in TuitionRules directory

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
feat(tuition rule): update default rule functionality

- admin can update the previously set default rule for all instructors

pswk1 committed:
Apply formatting changes

pswk1 committed:
fix(tuition rules): routing fix

- going back in the browser when viewing a single tuition rule will take admin back to viewing all current rules

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
refactor(tuition rules): rename components

- rename SetTuitionRule => ManageTopicTuition
- rename MainTuitionRulePage => ManageTuitionRule

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): component for rule edit history when history is empty

- use NoListAlert component when there are no edits made to the tuition rule

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): fetch policy in ManageTopicTuition

- set up Get Course Topic query and its fetch policy so that most recent changes/additions of default rules are reflected when admin is directed back to this component after creating/updating default rules

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): pagination and search

- admin can search for a topic by name
- topics are displayed fifteen topics per page

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): create instructor specific rule

- admin can create tuition rules specific to certain instructors

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): create instructor rule

- admin can create instructor-specific tuition rule after default tuition rule has been created

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
fix(tuition rules): fixes from feedback

- validation for creating rule of the same instructor or same combo of instructors
- "instructor-specific rule.." does not render when all default rules have been set
- select instructors width fixed
- spacing between private & small group tutoring
- Dedicated button for setting default tuition rule

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
fix(tuiton rule): validation

- fix instructor existing validation

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rules): validate update status when editing default rule

- admin is required to choose update status of edited default rule before submitting update

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): update instructor rule

update instructor rule

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes

pswk1 committed:
feat(tuition rule): retire modal

- clicking 'retire' button in instructor rule component opens a confirmation modal for retiring rule

pswk1 committed:
Merge branch 'tuition-rules' of github.com:omou-org/front-end into tuition-rules

pswk1 committed:
Apply formatting changes
